conjure-java-core/src/test/java/com/palantir/conjure/java/types/WireFormatTests.java

```diff
@@ -937,6 +937,12 @@ void testSealedInterfaceUnionType_deserialize_CamelCase() throws JsonProcessingE
                 .isEqualTo(CamelCaseUnion.camelCasedField("test"));
     }
 
+    @Test
+    void testSealedInterfaceUnionType_wireFormatBackCompatWithLegacyImpl() throws IOException {
+        assertThat(mapper.writeValueAsString(SimpleUnion.foo("foo")))
+                .isEqualTo(mapper.writeValueAsString(undertow.com.palantir.product.SimpleUnion.foo("foo")));
```

Comment:
It's not immediately clear which one is the sealed interface and which one isn't. Should we add some verification that one is sealed and one isn't? (to avoid test refactors breaking our assumptions by making both sealed or both unsealed)